homeassistant/components/miele/switch.py

```diff
@@ -0,0 +1,225 @@
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.DIALOG_OVEN,
+            MieleAppliance.STEAM_OVEN_MK2,
+        ),
+        description=MieleSwitchDescription(
+            key="poweronoff",
+            value_fn=lambda value: value.state_status,
+            off_value=1,
+            translation_key="power_on",
+            on_cmd_data={POWER_ON: True},
+            off_cmd_data={POWER_OFF: True},
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+    coordinator = config_entry.runtime_data
+
+    entities: list = []
+    entity_class: type[MieleSwitch]
+    for device_id, device in coordinator.data.devices.items():
+        for definition in SWITCH_TYPES:
+            if device.device_type in definition.types:
+                match definition.description.key:
+                    case "poweronoff":
+                        entity_class = MielePowerSwitch
+                    case "supercooling" | "superfreezing":
+                        entity_class = MieleSuperSwitch
+
+                entities.append(
+                    entity_class(coordinator, device_id, definition.description)
+                )
+    async_add_entities(entities)
+
+
+class MieleSwitch(MieleEntity, SwitchEntity):
+    """Representation of a Switch."""
+
+    entity_description: MieleSwitchDescription
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id,
+        description: MieleSwitchDescription,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator, device_id, description)
+        self.api = coordinator.api
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the device."""
+        await self.async_turn_switch(self.entity_description.on_cmd_data)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the device."""
+        await self.async_turn_switch(self.entity_description.off_cmd_data)
+
+    async def async_turn_switch(self, mode: dict[str, str | int | bool]) -> None:
+        """Set switch to mode."""
+        try:
+            await self.api.send_action(self._device_id, mode)
+        except aiohttp.ClientError as err:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="set_state_error",
+                translation_placeholders={
+                    "entity": self.entity_id,
+                },
+            ) from err
+
+
+class MielePowerSwitch(MieleSwitch):
+    """Representation of a power switch."""
+
+    entity_description: MieleSwitchDescription
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return the state of the switch."""
+        return self.coordinator.data.actions[self._device_id].power_off_enabled
```

Comment:
why don't we move this to an `is_on_fn`?

Reply:
There are only two implementaions of is_on in this platform and no more are planned. I think they are easy to read as they are. A lambda conctruction is not necessarily easier.